### PR TITLE
Feat: Add focus modal

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Add focus modal

**How** 
-  Add absolute element on top of everything with modal-like properties

**Testing**
- Add storybook stories `Molecules/Focus-modal`: 
  - `regular`
  - `overflow`

image

### CODE REVIEW

**Review by fPolic — APPROVED** [review]
Nice 👍 

Added a few comments to discuss potential reusability gains 😁

File: src/components/molecules/modal/focus-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    <div
+      className={clsx(
+        "absolute transition-all duration-100 inset-0 bg-grey-0 z-50 flex flex-col items-center",
+        {
+          "scale-[0.98] opacity-0": !isVisible,
+          "scale-100 opacity-100": isVisible,
+        }
+      )}
+    >
+      <FocusModalHeader
```
[the remark this answers is not in the document]
Comment (pKorsholm) [reply src/components/molecules/modal/focus-modal.tsx:44]: At the moment all occurrences in the design of a focus modal seem to be identical in structure, but I think in the future the flexibility you mention can be very useful to have baked in.

I think we should go for a hybrid in solution between the two, meaning we should export all the components like you describe here, as well as create a `BasicFocusModal` component which has an interface like the `FocusModal` in this PR and uses the `Header` and `Main` components to create a focus modal as it is used in the design now.

[the remark this answers is not in the document]
Comment (pKorsholm) [reply src/components/molecules/modal/focus-modal.tsx:44]: At the moment all occurrences in the design of a focus modal seem to be identical in structure, but I think in the future the flexibility you mention can be very useful to have baked in.

I think we should go for a hybrid in solution between the two, meaning we should export all the components like you describe here, as well as create a `BasicFocusModal` component which has an interface like the `FocusModal` in this PR and uses the `Header` and `Main` components to create a focus modal as it is used in the design now.

wdyt @fPolic?

File: src/components/molecules/modal/focus-modal.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+export default {
+  title: "Molecules/FocusModal",
+  component: FocusModal,
+} as ComponentMeta<typeof FocusModal>
+
+const Template: ComponentStory<typeof FocusModal> = (args) => {
+  const [open, setOpen] = useState(false)
+  return (
+    <div>
+      {open && (
```
[the remark this answers is not in the document]
Comment (pKorsholm) [reply src/components/molecules/modal/focus-modal.stories.tsx:16]: I think this is a great idea 😄  should we consider making this even more generalised and include the `in` and `out` animations in props as well, with the default being a fade and scale?

File: src/components/molecules/modal/focus-modal.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (fPolic) [reply src/components/molecules/modal/focus-modal.tsx:44]: Sounds good to me 👍

File: src/components/molecules/modal/focus-modal.stories.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (fPolic) [reply src/components/molecules/modal/focus-modal.stories.tsx:16]: That would be great :)

**Comment by srindom** [comment]
@fPolic do you wanna take another read through this before we merge?

**Review by fPolic — APPROVED** [review]

**Comment by fPolic** [comment]
> @fPolic do you wanna take another read through this before we merge?

@srindom LGTM!

**Pull request merged**